sklearn/multioutput.py

```diff
@@ -404,8 +404,22 @@ def fit(self, X, Y, sample_weight=None, **fit_params):
         self.classes_ = [estimator.classes_ for estimator in self.estimators_]
         return self
 
-    @property
-    def predict_proba(self):
+    def _check_predict_proba(self):
+        if hasattr(self, "estimators_") and not all(
+            [hasattr(estimator, "predict_proba") for estimator in self.estimators_]
+        ):
+            raise AttributeError(
+                "The base estimator should implement predict_proba method"
+            )
+        elif not hasattr(self, "estimator"):
+            raise AttributeError(
+                "predict_proba is not available because this"
+                f" {repr(self.__class__.__name__)} has no attribute 'estimator'"
+            )
+        return True
```

Comment:
I am still confused here. I would expect the following: we test all fitted estimators for the attribute. If the meta-estimator is not fitted, these fitted estimators are not existing, thus we test the base estimator.

```suggestion
        err_msg = "The base estimator should implement predict_proba method"
        if hasattr(self, "estimators_"):
            if not all(
                [hasattr(estimator, "predict_proba") for estimator in self.estimators_]
            ):
                raise AttributeError(err_msg)
            return True
        elif not hasattr(self.estimator, attr):
            raise AttributeError(err_msg)
        return True
```

Reply:
@glemaitre Thanks for the clarification. I've updated the code. Could you take a look at it again?